### PR TITLE
feat: Use external service for triggers

### PR DESCRIPTION
**Pull request by KevLehman**
## Issue(s)

[PROSVC-37](https://rocketchat.atlassian.net/browse/PROSVC-37)

## Further comments

[PROSVC-37]: https://rocketchat.atlassian.net/browse/PROSVC-37?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/meteor/tests/end-to-end/api/livechat/08-triggers.ts [hunk 1]
```diff
@@ -205,6 +206,191 @@ describe('LIVECHAT - triggers', function () {
...
+			await removePermissionFromAllRoles('view-livechat-manager');
+			await request
+				.post(api('livechat/triggers/external-service/test'))
+				.set(credentials)
+				.send({ webhookUrl: 'test', timeout: 1000, fallbackMessage: 'test', extraData: [] })
+				.expect(403);
+		});
+		it('should fail if Livechat_secret_token setting is empty', async () => {
+			await restorePermissionToRoles('view-livechat-manager');
+			await updateSetting('Livechat_secret_token', '');
```
Comment (MarcosSpessatto) [inline apps/meteor/tests/end-to-end/api/livechat/08-triggers.ts:350]: Can we put the default value back `after` all test cases?

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]

**Review by MarcosSpessatto — DISMISSED** [review]

File: apps/meteor/client/views/omnichannel/triggers/hooks/useActionFormFields.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import { SendMessageActionForm } from '../actions/SendMessageActionForm';
+
+type TriggerActions = 'send-message' | 'use-external-service';
+
+const actionForms = {
+	'send-message': SendMessageActionForm,
+	'use-external-service': ExternalServiceActionForm,
+} as const;
+
+export const useActionFormFields = (actionName: TriggerActions) => {
```
Comment (ggazzo) [inline apps/meteor/client/views/omnichannel/triggers/hooks/useActionFormFields.tsx:11]: transform this into a regular function

File: apps/meteor/app/livechat/server/api/lib/livechat.ts [hunk 3]
```diff
@@ -207,6 +208,7 @@ export async function settings({ businessUnit = '' }: { businessUnit?: string }
 		},
 		triggers,
 		departments,
+		premiumCapabilities: License.hasModule('livechat-enterprise'),
```
Comment (ggazzo) [inline apps/meteor/app/livechat/server/api/lib/livechat.ts:211]: I couldnt see this flag being used in any place

**Review by ggazzo — CHANGES_REQUESTED** [review]

File: apps/meteor/client/views/omnichannel/triggers/actions/ActionExternalServiceUrl.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+export const ActionExternalServiceUrl = ({ control, trigger, index, disabled, ...props }: ActionExternaServicelUrlType) => {
+	const t = useTranslation();
+
+	const serviceUrlFieldId = useUniqueId();
+	const serviceUrlFieldName = `actions.${index}.params.serviceUrl` as const;
+	const serviceTimeoutFieldName = `actions.${index}.params.serviceTimeout` as const;
+
+	const serviceTimeoutValue = useWatch({ control, name: serviceTimeoutFieldName });
+	const [serviceUrlError] = useFieldError({ control, name: serviceUrlFieldName });
+	const [isSuccessMessageVisible, setSuccessMessageVisible] = useState(false);
```
Comment (ggazzo) [inline apps/meteor/client/views/omnichannel/triggers/actions/ActionExternalServiceUrl.tsx:32]: use useSafely

Comment (ggazzo) [inline apps/meteor/client/views/omnichannel/triggers/actions/ActionExternalServiceUrl.tsx:32]: const isSuccessMessageVisible = useDebouncedValue(webhookTest.isLoading, 10000);

Comment (aleksandernsilva) [reply apps/meteor/client/views/omnichannel/triggers/actions/ActionExternalServiceUrl.tsx]: I tried a few variations but unfortunately this did not work as the scenario here is a bit different than the one we saw together.

**Review by MarcosSpessatto — DISMISSED** [review]

**Review by ggazzo — APPROVED** [review]

**Review by ggazzo — DISMISSED** [review]

**Review by MarcosSpessatto — APPROVED** [review]

**Review by ggazzo — DISMISSED** [review]

**Review by aleksandernsilva — APPROVED** [review]

**Review by MarcosSpessatto — DISMISSED** [review]

**Review by ggazzo — APPROVED** [review]

**Review by aleksandernsilva — APPROVED** [review]

**Review by MarcosSpessatto — APPROVED** [review]

**Pull request merged**